src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs

```diff
@@ -942,17 +972,29 @@ private static async Task<DurableOrchestrationStatus> GetDurableOrchestrationSta
             return this.httpApiHandler.CreateCheckStatusResponse(request, instanceId, attribute, returnInternalServerErrorOnFailure);
         }
 
-        private static void TrackNameAndScheduledTime(JObject historyItem, EventType eventType, int index, Dictionary<string, EventIndexDateMapping> eventMapper)
+        private static void TrackNameAndScheduledTime(JObject historyItem, EventType eventType, int index, Dictionary<string, EventIndexDateMapping> eventMapper, bool showHistoryInput)
         {
-            eventMapper.Add($"{eventType}_{historyItem["EventId"]}", new EventIndexDateMapping { Index = index, Name = (string)historyItem["Name"], Date = (DateTime)historyItem["Timestamp"] });
+            if (showHistoryInput)
```

Comment:
Correct me if I'm wrong but I think we can always add Input to this mapping and it wont actually affect what is displayed for the customer. So we shouldn't need to change behavior for showHistoryInput and can leave this method with input always being mapped. It should just be null if there isn't a value.